src/main/scala/wdl4s/Import.scala

```diff
@@ -1,17 +1,21 @@
 package wdl4s
 
+import better.files.File
 import wdl4s.AstTools.EnhancedAstNode
 import wdl4s.parser.WdlParser.{Terminal, Ast, AstNode}
 
 object Import {
   def apply(astNode: AstNode): Import = {
     val uri = astNode.asInstanceOf[Ast].getAttribute("uri").sourceString
-    val importNamespace = Option(astNode.asInstanceOf[Ast].getAttribute("namespace")).map(_.asInstanceOf[Terminal])
-    Import(uri, importNamespace)
+    val namespaceTerminal = Option(astNode.asInstanceOf[Ast].getAttribute("namespace")).map(_.asInstanceOf[Terminal]).getOrElse {
+      astNode.asInstanceOf[Ast].getAttribute("uri").asInstanceOf[Terminal]
+    }
+
+    Import(uri, namespaceTerminal)
   }
 }
 
 // FIXME: I dislike dragging the AST along but it's necessary for "compile" time error syntax highlighting, argh
-case class Import(uri: String, namespaceTerminal: Option[Terminal]) {
-  val namespaceName: Option[String] = namespaceTerminal.map(_.sourceString)
+case class Import(uri: String, namespaceTerminal: Terminal) {
+  val namespaceName: String = File(namespaceTerminal.sourceString).nameWithoutExtension
```

Comment:
If the `namespaceTerminal` comes from the `as` clause you probably don't want/need to do `File(namespaceTerminal.sourceString).nameWithoutExtension`, although I imagine in most cases it wouldn't affect the alias string anyway.